pkg/tfbridge/walk.go

```diff
@@ -219,8 +229,16 @@ func schemaPathToPropertyPathInner(
 			basePath = append(basePath, "*")
 		}
 		switch e := schema.Elem().(type) {
+
+		// (schema, schemaInfo) represents a list or set nested object (schema ~
+		// List[{x: T}] or schema ~ Set[{x: T}]), so we traverse the inner object
+		// with the associated .Elem.Fields.
 		case shim.Resource: // object element type
-			return schemaPathToPropertyPath(basePath, schemaPath[1:], e.Schema(), schemaInfo.Fields)
+			elem := schemaInfo.Elem
```

Comment:
Does this also need a test? Looks like the tests only cover the other direction.

Reply:
The test is bi-directional. Both changes are covered.

---

I actually only noticed that this needed to be fixed because the test I added was still failing.